testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -193,19 +194,257 @@ def process_test_clean(test_tag, work_dir):  # {{{
+    local_parallel_code += "\n\n\n"
+    local_parallel_code += "iteration = 0\n"
+    local_parallel_code += "compleated = []\n"
+    local_parallel_code += "failed = []\n"
+    local_parallel_code += "running = []\n"
+    local_parallel_code += "adding_phase = True\n"
+    local_parallel_code += "processing_phase = False\n"
+    local_parallel_code += "done = False\n"
+    local_parallel_code += "start_time = time.time()"
+    local_parallel_code += "\n"
+    local_parallel_code += "while True:\n"
+    local_parallel_code += "    if len(compleated) + len(failed) < len(testcase_data.keys()):\n"
+    local_parallel_code += "        adding_phase = True\n"
+    local_parallel_code += "    else:\n"
+    local_parallel_code += "        adding_phase = False\n"
+    local_parallel_code += "        done = True\n"
+    local_parallel_code += "    if adding_phase:\n"
+    local_parallel_code += "        running_names = [running_proc[1] for running_proc in running]\n"
+    local_parallel_code += "        for key in testcase_data.keys():\n"
+    local_parallel_code += "            if not key in compleated and not key in running_names:\n"
+    local_parallel_code += "                if not testcase_data[key]['prereqs'] == [] and testcase_data[key]['runnable'] == False:\n"
+    local_parallel_code += "                    prereq_tests = [prereq['name'] for prereq in testcase_data[key]['prereqs']]\n"
+    local_parallel_code += "                    prereq_compleated =  all(prereq in compleated for prereq in prereq_tests)\n"
+    local_parallel_code += "                    if prereq_compleated:\n"
+    local_parallel_code += "                        testcase_data[key]['runnable'] = True\n"
+    local_parallel_code += "                if number_of_procs >= testcase_data[key]['procs'] and testcase_data[key]['runnable'] == True:\n"
+    local_parallel_code += "                    number_of_procs = number_of_procs - testcase_data[key]['procs']\n"
+    local_parallel_code += "                    process , key = run(key)\n"
+    local_parallel_code += "                    running.append((process,key))\n"
+    local_parallel_code += "                    print('we have enough to add: {}, {}, {}'.format(testcase_data[key]['procs'], number_of_procs, key))\n"
+    local_parallel_code += "                else:\n"
+    local_parallel_code += "                    processing_phase = True\n"
+    local_parallel_code += "                    adding_phase = False \n"
+    local_parallel_code += "    if processing_phase:\n"
+    local_parallel_code += "        for process in running:\n"
+    local_parallel_code += "            pid = process[0].pid\n"
+    local_parallel_code += "            running_key = process[1]\n"
+    local_parallel_code += "            if not psutil.pid_exists(pid):\n"
+    local_parallel_code += "                error_out = subprocess.call(['grep', 'FAIL', running_key.replace(' ', '_')], stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)\n"
```

Comment:
This checks to see if the run failed and what to do with it

Reply:
I don't think this `grep` call is working as you want.  And why not use python directly rather than a subprocess call?

Reply:
Following up a bit:
``` python
filename = running_key.replace(' ', '_')
with open(filename) as f:
    error_out = 'FAIL' in f.read()
```
should accomplish the same thing without a subprocess call.